### ISSUE TITLE
defining a local_resource with the same name as another resource should be an error

### ISSUE THREAD

**Issue by landism** [tilt-dev/tilt#2919]
repro:
Create a Tiltfile and `tilt up` the following:
```
local_resource('foo', 'echo foo')
local_resource('foo', 'echo foo')
```

observed:
no errors, and two resources named 'foo'

expected:
an error that 'foo' was already defined

context:
a lot of Tilt code relies on the constraint that resource names are unique, and local_resource apparently allows a Tiltfile to defy that

solution notes:
* ideally when we learn of the conflict, the error can specify where each of the conflicting names is coming from (the report that led to this was the result of a tiltfile loading a tiltfile loading a tiltfile)
* we need some unification of different types of resources in tiltfile execution (currently k8s resources check for conflicts with other k8s resources, but not local resources, and local resources don't check for conflicts at all). for the purposes of this bug, this could be simple as adding a `map[model.ManifestName]syntax.Position` to `tiltfileState`.
* k8s resources aren't actually constructed until the end of tiltfile execution, which might complicate things

**Comment by maiamcc** [comment]
Could we feed this and #1852 with one scone?

**Comment by abdullahzameek** [comment]
Resolved with PR #3490
